workspaces/core/src/select/Select.tsx

```diff
@@ -0,0 +1,150 @@
+import {determineMenuBackgroundColor} from "./private/getBackgroundColor";
+import {getWrapperClasses} from "./private/getWrapperClasses";
+import Option from "./private/types/OptionProps";
+import SelectProps from "./private/types/SelectProps";
+import useEventListeners from "./private/useEventListeners";
+
+const Select = forwardRef<HTMLDivElement, SelectProps>(({
+  options,
+  label,
+  value,
+  disabled,
+  isError,
+  placeholderSlot,
+  placeholderValue,
+  hintSlot,
+  onChange,
+  size = 'md',
+  amountOfVisibleItems = 9999,
+  menuWidth,
+  headerSlot,
+  footerSlot,
+  leftSlot,
+  isSharpLeftSide,
+  isSharpRightSide,
+  isSharpTopSide,
+  isSharpBottomSide,
+  isTopBottomBorderHidden,
+  isSideBorderHidden,
+  isRtl,
+  isSearchable,
+  noResultsMessage,
+  formatOptionLabel
+}) => {
+  const [hoveredIndex, setHoveredIndex] = useState(-1);
+  const [selectedIndex, setSelectedIndex] = useState(-1);
+  const placeholder = value ? options?.filter((option) => option.value === value)[0]?.label
+    : placeholderSlot ?? placeholderValue ?? 'Choose an option';
+  const {
+    menuOpen,
+    search,
+    menuOptions,
+    inputFocused,
+    inputRef,
+    selectRef,
+    menuRef,
+    onSelectClick,
+    onInputFocus,
+    selectMenuItem,
+    handleInputKeydown
+  } = useEventListeners({
+    options,
+    onChange,
+    value,
+    hoveredIndex,
+    selectedIndex,
+    isSearchable,
+    disabled,
+    amountOfVisibleItems,
+    setHoveredIndex,
+    setSelectedIndex
+  });
+
+  const classNames = getWrapperClasses({
+    size,
+    disabled,
+    isSharpTopSide,
+    isSharpLeftSide,
+    isSharpRightSide,
+    isSharpBottomSide,
+    isTopBottomBorderHidden,
+    isSideBorderHidden,
+    isRtl,
+    isError,
+    inputFocused
+  });
+
+  return (<div className={`support-colors flex flex-col text-popo ${ disabled ? 'opacity-30' : ''}` }>
+    { !!label && size !== 'xl' && (<div className='text-base mb-2'>{label}</div>) }
+
+    <div
+      className={classNames}
+      onClick={onSelectClick}
+      ref={selectRef}
+    >
+      <input
+        className='bg-gohan text-transparent z-[1] focus:outline-none focus:border-none'
+        disabled={disabled}
+        ref={inputRef}
+        onKeyDown={(e) => handleInputKeydown(e)}
+        onFocus={() => onInputFocus(true)}
+        onBlur={() => onInputFocus()}
+        defaultValue={isSearchable ? search : menuOptions.find((option: Option) => option.value === value)?.label || ''}
+        readOnly={!isSearchable}
+      />
+
+      {!!leftSlot && (<div className={`absolute left-0 top-0 w-6 h-6 flex justify-center items-center overflow-hidden z-10 m-${size === 'xl' ? 4 : 2}`}>{leftSlot}</div>)}
+
+      <div className={`absolute w-full flex items-center justify-between z-[2] ${ disabled ? 'cursor-not-allowed' : 'cursor-pointer' } ${!value ? 'text-trunks' : ''} ${leftSlot ? 'pl-6' : ''} ${isRtl ? `pl-${ size === 'md' ? 3 : 4 }` : `pr-${ size === 'md' ? 3 : 4 }`}`}>
```

Comment:
`${isRtl ? `pl-${ size === 'md' ? 3 : 4 }` : `pr-${ size === 'md' ? 3 : 4 }`}`}`
Please don't use string concatenation to create class names there.